File: backend/app/schemas/part_types.py
from __future__ import annotations

from typing import Any

from pydantic import BaseModel


class PartTypeFieldResponse(BaseModel):
    id: int
    field_key: str
    label: str
    field_type: str
    is_required: bool
    sort_order: int
    options: list[Any] | dict[str, Any] | None = None
    default_unit: str | None = None
    help_text: str | None = None


class PartTypeResponse(BaseModel):
    id: int
    name: str
    slug: str
    description: str | None = None
    is_builtin: bool
    is_active: bool
    template_version: int
    field_count: int
    fields: list[PartTypeFieldResponse]


class PartTypeCollectionResponse(BaseModel):
    total: int
    builtin_count: int
    custom_count: int
    total_fields: int
    part_types: list[PartTypeResponse]

File: frontend/src/types/partTypes.ts
export interface PartTypeField {
  id: number;
  field_key: string;
  label: string;
  field_type: string;
  is_required: boolean;
  sort_order: number;
  options: unknown[] | Record<string, unknown> | null;
  default_unit: string | null;
  help_text: string | null;
}

export interface PartType {
  id: number;
  name: string;
  slug: string;
  description: string | null;
  is_builtin: boolean;
  is_active: boolean;
  template_version: number;
  field_count: number;
  fields: PartTypeField[];
}

export interface PartTypeCollection {
  total: number;
  builtin_count: number;
  custom_count: number;
  total_fields: number;
  part_types: PartType[];
}
